src/librustdoc/html/render/mod.rs

```diff
@@ -115,7 +114,7 @@ crate struct Context<'tcx> {
     /// Tracks section IDs for `Deref` targets so they match in both the main
     /// body and the sidebar.
     deref_id_map: Rc<RefCell<FxHashMap<DefId, String>>>,
-    crate shared: Arc<SharedContext<'tcx>>,
+    crate shared: SharedContext<'tcx>,
```

Comment:
This is likely what broke the tests. Before, the same context was shared between all clones; now it makes a copy. You'll need to remove all clones of `SharedContext` for this to have the same behavior. I'd suggest dropping this commit for now.